src/TLuaInterpreter.cpp

```diff
@@ -13723,3 +13439,19 @@ int TLuaInterpreter::startPermSubstringTrigger(const QString & name, const QStri
 
 }
 
+static int host_key = 0;
+
+static void store_host_in_lua_state(lua_State* L, Host* h) {
```

Comment:
Not only do I think it works, I know it works.

The registry is explained: https://www.lua.org/manual/5.1/manual.html#3.5

Just run:

```lua
for k, v in pairs(debug.getregistry()) do print(k, v) end
```

Both before and after this change.